api/v1alpha4/cluster_webhook.go

```diff
@@ -86,6 +103,111 @@ func (c *Cluster) validate() error {
 		)
 	}
 
+	// If the Cluster uses a managed topology.
+	if c.Spec.Topology != nil {
```

Comment:
Could we move everything inside this validation into a `validateTopology` function instead?

Reply:
done